Extension/src/LanguageServer/settings.ts

```diff
@@ -414,337 +432,299 @@ export class CppSettings extends Settings {
-        return super.Section.get<boolean>("vcFormat.space.groupSquareBrackets") === true;
-    }
+        if (allowNull && value === null) {
+            return null;
+        }
 
-    public get vcFormatSpaceWithinLambdaBrackets(): boolean {
-        return super.Section.get<boolean>("vcFormat.space.withinLambdaBrackets") === true;
+        if (isBoolean(value)) {
+            return value;
+        }
+        return setting.default;
     }
 
-    public get vcFormatSpaceBetweenEmptyLambdaBrackets(): boolean {
-        return super.Section.get<boolean>("vcFormat.space.betweenEmptyLambdaBrackets") === true;
-    }
+    // Returns the value of a setting as a string with proper type validation and checks for valid enum values.
+    private getAsString(settingName: string): string;
+    private getAsString(settingName: string, allowNull: boolean): string | null;
+    private getAsString(settingName: string, allowNull: boolean = false): string | null {
+        const value: any = super.Section.get<any>(settingName);
+        const setting = getRawSetting("C_Cpp." + settingName);
+        if (!isString(setting.default) && !(allowNull && setting.default === null)) {
+            console.error(`Default value for ${settingName} is expected to be string or null.`);
+        }
 
-    public get vcFormatSpaceBeforeComma(): boolean {
-        return super.Section.get<boolean>("vcFormat.space.beforeComma") === true;
-    }
+        if (allowNull && value === null) {
+            return null;
+        }
 
-    public get vcFormatSpaceAfterComma(): boolean {
-        return super.Section.get<boolean>("vcFormat.space.afterComma") === true;
+        if (setting.enum !== undefined) {
+            if (settingName === "loggingLevel" && isString(value) && isNumber(Number(value)) && Number(value) >= 0) {
+                return value;
+            }
+            if (this.isValidEnum(setting.enum, value)) {
+                return value;
+            }
+        } else if (isString(value)) {
+            return value;
+        }
+        return setting.default as string;
     }
 
-    public get vcFormatSpaceRemoveAroundMemberOperators(): boolean {
-        return super.Section.get<boolean>("vcFormat.space.removeAroundMemberOperators") === true;
-    }
+    // Returns the value of a setting as a number with proper type validation and checks if value falls within the specified range.
+    private getAsNumber(settingName: string): number;
+    private getAsNumber(settingName: string, allowNull: boolean): number | null;
+    private getAsNumber(settingName: string, allowNull: boolean = false): number | null {
+        const value: any = super.Section.get<any>(settingName);
+        const setting = getRawSetting("C_Cpp." + settingName);
+        if (!isNumber(setting.default) && !(allowNull && setting.default === null)) {
+            console.error(`Default value for ${settingName} is expected to be number or null.`);
```

Comment:
Why not use something like ` number ${allowNull ? " or null" : ""}`?

Reply:
This is really just for us to see, so I didn't put that much thought into it. I can change it though.

Reply:
I wanted to avoid any possible confusing around the "or null" in case someone hits it.

Reply:
I am fixing it.